folly/FixedString.h

```diff
@@ -1110,22 +1110,16 @@ class BasicFixedString : private detail::fixedstring::FixedStringBase {
    * \return `*(data() + i)`
    */
   constexpr Char& operator[](std::size_t i) noexcept {
-#ifdef NDEBUG
-    return data_[i];
-#else
-    return data_[detail::fixedstring::checkOverflow(i, size_)];
-#endif
+    return kIsDebug ? data_[detail::fixedstring::checkOverflow(i, size_)] :
```

Comment:
C++11 constexpr rules are no longer required, so:

```
i = kIsDebug ? detail::fixedstring::checkOverflow(i, size_) : i;
return data_[i];
```

Or even just define `checkOverflowIfDebug` and:

```
return data_[detail::fixedstring::checkOverflowIfDebug(i, size_)];
```

Here and below.